pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -214,9 +214,15 @@ func (s *DiscoveryServer) ClearCacheFunc() func() {
 		// TODO: propagate K8S version and use it instead
 		versionMutex.Lock()
 		s.env.PushContext = push
-		version = time.Now().Format(time.RFC3339)
+		if err = s.ConfigGenerator.BuildSharedPushState(s.env); err == nil {
```

Comment:
We have quite a bit of initialization - this we want to not increment the version, it should be in case of any error, and it should prevent pushing.

Reply:
I don't understand.